src/java/org/apache/cassandra/index/sai/plan/QueryController.java

```diff
@@ -97,8 +98,11 @@ public class QueryController implements Plan.Executor
      * 0 disables the optimizer.
      * 1 enables the optimizer and tells the optimizer to respect the intersection clause limit.
      * Higher values enable the optimizer and disable the hard intersection clause limit.
+     * Note: the config is not final to simplify testing. It is initialized before other final objects in this class,
```

Comment:
Thanks, I didn't have that as part of my mental model... I see it here too https://shipilev.net/blog/2014/safe-public-construction/. I'll remove the comment since it's not reliant on other fields being final.